Automattic/woocommerce-payments
# Fix flash of WooCommerce Payments gateway title name.

Fixes #2891 

#### Changes proposed in this Pull Request

* Add filter for `woocommerce_gateway_title` to call `maybe_filter_gateway_title`.
* `maybe_filter_gateway_title` checks to see if it's our gateway and default title being used, if so it then assigns the new title with the value of `checkout_title` (Popular payment methods). From there it checks to see if there is 1 available method, and it will assign that method's title if so. If there are 0 methods available, it assigns the `card` method's title.

#### Testing instructions

* Make sure UPE is enabled.
* Add card and 1 other UPE method.
* Add item to cart and go to checkout.
* Use Multi-Currency widget and set currency to EUR.
* When the page reloads, the `Popular payment methods` title should show for the payment method, without a flash of WooCommerce Payments.
* Go to Payments > Settings and remove the card method, leaving a UPE method.
* Go back to checkout.
* When the page reloads, the payment title should show the name of the method you selected, without a flash of WooCommerce Payments.
* Go to Payments > Settings and add the card method back also leaving a UPE method.
* Go back to checkout.
* Use Multi-Currency widget and set currency to USD.
* When the page reloads, the `Credit card / debit card` title should show for the payment method, without a flash of WooCommerce Payments. Other UPE method should not be available.


-------------------


**Post merge**
